contracts/ClearingHouse.sol

```diff
@@ -848,30 +851,29 @@ contract ClearingHouse is
             0
         );
 
-        if (response.pnlToBeRealized != 0) {
-            // if realized pnl is not zero, that means trader is reducing or closing position
-            // trader cannot reduce/close position if the remaining account value is less than
-            // accountValue * LiquidationPenaltyRatio, which
-            // enforces traders to keep LiquidationPenaltyRatio of accountValue to
-            // shore the remaining positions and make sure traders having enough money to pay liquidation penalty.
+        int256 takerPositionSizeAfterSwap =
+            IAccountBalance(_accountBalance).getTakerPositionSize(params.trader, params.baseToken);
 
-            // CH_NEMRM : not enough minimum required margin after reducing/closing position
-            require(
-                getAccountValue(params.trader) >=
-                    _getTotalAbsPositionValue(params.trader).mulRatio(_getLiquidationPenaltyRatio()).toInt256(),
-                "CH_NEMRM"
-            );
-        }
+        bool hasBecameInversePosition =
+            (takerPositionSizeBeforeSwap < 0 && takerPositionSizeAfterSwap > 0) ||
+                (takerPositionSizeBeforeSwap > 0 && takerPositionSizeAfterSwap < 0)
+                ? true
+                : false;
```

Comment:
we can use bit xor for this, 

```
bool hasBecameInversePosition = (takerPositionSizeBeforeSwap ^ takerPositionSizeAfterSwap) < 0;
```

the first bit after xor would always be 1 if  `hasBecameInversePosition` and `takerPositionSizeAfterSwap` are in different sign.